graph/number_of_islands_dfs.cpp

```diff
@@ -0,0 +1,60 @@
+#include <algorithm>
+#include <cmath>
+#include <iostream>
+#include <map>
+#include <numeric>
+#include <set>
+#include <string>
+#include <unordered_map>
+#include <vector>
+//#include <bits/stdc++.h>
+using namespace std;
+typedef long long ll;
+#define int int64_t
+
+void dfs(vector<vector<char>> &matrix, int row, int col) {
+    if (row < 0 || row >= matrix.size() || col < 0 || col >= matrix[0].size()) {
+        return;
+    }
+    if (matrix[row][col] == '0') {
+        return;
+    }
+    matrix[row][col] = '0';
+    dfs(matrix, row - 1, col);
+    dfs(matrix, row + 1, col);
+    dfs(matrix, row, col + 1);
+    dfs(matrix, row, col - 1);
+}
+
+int32_t main() {
+    ios_base::sync_with_stdio(false);
+    cin.tie(0);
+    cout.tie(0);
```

Comment:
This is not competitive programing. So do not use this here 😄

Reply:
Okay 😅